flask/app.py

```diff
@@ -1982,6 +1997,10 @@ def wsgi_app(self, environ, start_response):
                                a list of headers and an optional
                                exception context to start the response
         """
+        with self._before_request_lock:
+            if not self._got_first_request and self._should_add_static_route:
+                self._add_static_route()
+                self._should_add_static_route = False
```

Comment:
Why do we need this variable? We can just check `if self._got_first_request and self.has_static_folder: ...` 

I also don't understand why it's `not self._got_first_request` here.